src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -555,19 +538,7 @@ public DetailedPackageMetadata PackageMetadata
                         NuGetUIThreadHelper.JoinableTaskFactory
                             .RunAsync(async () =>
                             {
-                                try
-                                {
-                                    var readMe = await GetReadMeMD();
-                                    await ReadMePreviewViewModel.UpdateMarkdownAsync(readMe);
-                                    IsReadMeAvailable = !string.IsNullOrWhiteSpace(readMe);
-                                }
-                                catch (Exception)
-                                {
-                                    await ReadMePreviewViewModel.UpdateMarkdownAsync("");
-                                    ReadMePreviewViewModel.IsErrorWithReadMe = true;
-                                    IsReadMeAvailable = true;
-                                    throw;
-                                }
+                                await ReadMePreviewViewModel.UpdateReadMe(_packageMetadata?.PackagePath, _packageMetadata?.Id);
```

Comment:
Could the Readme view call on update readme rather than the setter of the PackageMetadata? It looks like the data context is being set on the view to the new ReadmePreviewViewModel already, so is that a good time to update the readme which will invoke a property changed event on the VM and the view can react by then rendering the markdown? This would remove the responsibility from the DetailControlModel of having to drive the ReadMeViewModel.